test/Microsoft.AspNet.Mvc.IntegrationTests/ModelPrefixSelectionIntegrationTest.cs

```diff
@@ -0,0 +1,394 @@
+        }
+
+        [Fact]
+        public async Task ComplexModel_PrefixSelected_ByValueProviderValue_WithFilteredValueProviders()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Person2),
+                BindingInfo = new BindingInfo()
+                {
+                    BindingSource = BindingSource.Query,
+                },
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request =>
+            {
+                // This will cause selection of the "parameter" prefix.
+                request.QueryString = new QueryString("?parameter=");
+
+                // This value won't be used, because we select the "parameter" prefix.
+                request.Form = new FormCollection(new Dictionary<string, StringValues>()
+                {
+                    { "Name", "Billy" },
+                });
+            });
+
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+            Assert.Equal("parameter", modelBindingResult.Key);
+
+            var model = Assert.IsType<Person2>(modelBindingResult.Model);
+            Assert.Null(model.Name);
+
+            Assert.Equal(0, modelState.Count);
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        private class Person3
+        {
+            [FromForm]
+            public string Name { get; set; }
+        }
+
+        [Fact]
+        public async Task ComplexModel_EmptyPrefixSelected_NoMatchingValueProviderValue()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Person3),
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request =>
+            {
+                // This can't be used because of [FromForm] on the property.
+                request.QueryString = new QueryString("?Name=");
+            });
+
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+            Assert.Equal(string.Empty, modelBindingResult.Key);
```

Comment:
I don't think that's an improvement. `Assert.Empty` only has meaning because string happens to implement IEnumerable. `Assert.Equal(string.Empty, modelBindingResult.Key);` is clearer